src/Models/Behaviors/HasNesting.php

```diff
@@ -93,4 +94,12 @@ public static function flattenTree(array $nodeTree, int $parentId = null)
 
         return $nodeArrays;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function newCollection(array $models = [])
+    {
+        return new Collection($models);
```

Comment:
I think here we do not need to Alias it and we can just return new NestedsetCollection